### PR TITLE
Fix calculation of timeout before next change run

### PR DESCRIPTION
**Pull request by a1exsh**
1. Changes is not a list, but a mapping from name to list of `Change` objects.
2. Change with a long timeout should not postpone run of scheduled checks.
3. A newly added change has to be executed without delay caused by other changes.

Internal ticket: 789

## Review
- [ ] Tests
- [ ] Documentation
- [ ] CHANGELOG

### CODE REVIEW

File: bubuku/controller.py [hunk 1]
```diff
@@ -192,5 +203,6 @@ def _add_change_to_queue(self, change):
     def stop(self, change: Change):
         _LOG.info('Stopping controller with additional change: {}'.format(change.get_name() if change else None))
         # clear all pending changes
```
Comment (a1exsh) [inline bubuku/controller.py:205]: Was this a `TODO` comment?  I don't see that we clear anything here.

File: bubuku/controller.py [hunk 2]
```diff
@@ -133,6 +135,11 @@ def _run_changes(self, running_changes: dict) -> list:
                             changes_to_remove.append(change.get_name())
                         else:
                             _LOG.info('Action {} will be executed on next loop step'.format(change))
+                            time_till_next_run = change.time_till_next_run()
+                            if min_time_till_next_change_run is None:
+                                min_time_till_next_change_run = time_till_next_run
+                            else:
```
Comment (antban) [inline bubuku/controller.py:141]:
```suggestion
                            elif time_till_next_run is not None:
```

Comment (antban) [reply bubuku/controller.py:141]: but it's nitpicking feel free to merge as it is.

File: bubuku/controller.py [hunk 3]
```diff
@@ -192,5 +203,6 @@ def _add_change_to_queue(self, change):
     def stop(self, change: Change):
```
Comment (antban) [inline bubuku/controller.py:203]:
```suggestion
    def stop(self, change: Optional[Change]):
```

File: bubuku/controller.py (hunk 1 above)
Comment (antban) [reply bubuku/controller.py:205]: I think it should be removed

**Review by antban — APPROVED** [review]
I'm fine as it is right now

File: bubuku/controller.py (hunk 3 above)
Comment (antban) [reply bubuku/controller.py:203]: nitpicking again

File: bubuku/controller.py (hunk 2 above)
Comment (a1exsh) [reply bubuku/controller.py:141]: What is semantics of `None`?  The method is not even supposed to return that (not optional).

**Comment by a1exsh** [comment]
:+1:

**Comment by antban** [comment]
:+1:

**Pull request merged**